daft/dataframe/lance_write_sink.py

```diff
@@ -0,0 +1,102 @@
+import pathlib
+from typing import TYPE_CHECKING, Literal, Optional, Union
+
+from daft.context import get_context
+from daft.daft import IOConfig
+from daft.dataframe.dataframe import DataFrame, WriteSink
+
+if TYPE_CHECKING:
+    import lance
+
+
+class LanceWriteSink(WriteSink[list["lance.fragment.FragmentMetadata"]]):
+    """WriteSink for writing data to a Lance dataset."""
+
+    def __init__(
+        self,
+        uri: Union[str, pathlib.Path],
+        mode: Literal["create", "append", "overwrite"],
+        io_config: Optional[IOConfig] = None,
+    ):
```

Comment:
notes from discussion: we discussed that write configuration (write location, bucketing strategy, write mode e.t.c) should go into WriteSink initialization which aligns pretty well with this exmaple sink implementation.